source/agora/consensus/EnrollmentManager.d

```diff
@@ -577,9 +577,8 @@ public class EnrollmentManager
         foreach (const ref key; keys)
         {
             const preimage = this.validator_set.getPreimageAt(key, height);
-            // this should not happen. validators which didn't reveal the
-            // preimage should not be in the active validator set 'keys'.
-            assert(preimage != PreImageInfo.init);
```

Comment:
This should be restored, which will be dealt with in another PR. Currently, this makes many assertion failures due to timing issues which means that the function is called before the right pre-images are not revealed in many test cases.